### DISCUSSION TITLE
advection diffusion algorithm

### DISCUSSION THREAD

**Discussion by Devin-one** [PMEAL/OpenPNM#2598] [Q&A]
Hello professor, I want to learn the advection diffusion algorithm recently. But I don't seem to find any examples about it. Do you have any examples or papers on it？

**Comment by jgostick** [comment]
Hi @Liqiwen123 
We have a few things you can refer to, including our [new documentation](https://openpnm.org/examples/applications/dispersion_coefficient.html). We had a graduate student here at UW working on this for his thesis so you can expect a paper in the next few months.

**Reply by Devin-one** [reply]
Thank you for your suggestion. I have been studying your example recently. My research purpose is to get the velocity field and pressure field, so I don't know whether I should set it like the Stokes flow algorithm in the example. By the way, what are A and b in the coefficient matrix in convection diffusion? Can x only represent the pore concentration? Thank you in advance for your reply.
![图片](https://user-images.githubusercontent.com/93040662/196980757-a18c5738-3645-4d71-b426-0df393f55422.png)

**Reply by jgostick** [reply]
The StokesFlow is solved first and then the flow is "set", the advection diffusion algorithm is basically a diffusion algorithm where the local velocity of the fluid is included in the diffusive conductance.  So yes, x only represents concentration.  If you want to know the pressure and fluid velocity, you ask the StokesFlow algorithm.
